staging/src/k8s.io/kubectl/pkg/cmd/version/version.go

```diff
@@ -94,7 +93,7 @@ func (o *Options) Complete(f cmdutil.Factory, cmd *cobra.Command) error {
 	o.discoveryClient, err = f.ToDiscoveryClient()
 	// if we had an empty rest.Config, continue and just print out client information.
 	// if we had an error other than being unable to build a rest.Config, fail.
-	if err != nil && !clientcmd.IsEmptyConfig(err) {
+	if err != nil && err != genericclioptions.ErrEmptyConfig {
```

Comment:
should we fix the  IsEmptyConfig function instead? This also needs a test demonstrating the bug to prevent another regression

Reply:
I don't think change the fucntion is a good idea since this function is used in blow:
```
func (f *ConfigFlags) ToRESTConfig() (*rest.Config, error) {
	return f.ToRawKubeConfigLoader().ClientConfig()
	config, err := f.ToRawKubeConfigLoader().ClientConfig()
	// replace client-go's ErrEmptyConfig error with our custom, more verbose version
	if clientcmd.IsEmptyConfig(err) {
		return nil, ErrEmptyConfig
	}
	return config, err
}
```
But add the test is a good idea, I'll do it.

Reply:
@liggitt Test case is done. At first I want to wirte test in `test/cmd/version.sh`, but I realize the condition should be `kubectl version` without config file. But all those test cases are initialized with the config file so I create `version_test.go` to write this special test case.

Reply:
it still seems like `clientcmd.IsEmptyConfig` should return true if it previously returned true for an empty configuration

Reply:
I agree. Ok, I'll try to change this.